enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -5783,48 +6017,80 @@ class AdjointGenerator
         Value *orig_op = call.getOperand(4);
         Value *orig_comm = call.getOperand(5);
 
-        bool isSum = false;
-        if (Constant *C = dyn_cast<Constant>(orig_op)) {
-          while (ConstantExpr *CE = dyn_cast<ConstantExpr>(C)) {
-            C = CE->getOperand(0);
-          }
-          if (auto GV = dyn_cast<GlobalVariable>(C)) {
-            if (GV->getName() == "ompi_mpi_op_sum") {
-              isSum = true;
+        if (!forwardMode) {
+          bool isSum = false;
```

Comment:
Same sum comment

Reply:
we are now catching non-sum ops, just like in reverse mode

Reply:
we are now catching non-sum ops, just like in reverse mode